tests/test_analyze__main.py

```diff
@@ -0,0 +1,112 @@
+# tests/test_analyze__main.py
+#
+# These are end-to-end tests that run module `birdnet_analyzer.analyze`
+# as if calling it from the command-line.
+#
+# When run with certain arguments on known example input,
+# an output file should be generated with expected specific format and data
+# as found in tests/resources/SNAPSHOT.analyze__main.*)
+#
+# How to run these tests;
+# 1. (Prerequisite) from root of this repository run: `pip install -r requirements.txt`
+# 2. From root of this repository run: `pytest tests/test_analyze__main.py`
```

Comment:
I originally started adding a `DEVELOPMENT.adoc` readme, not quite knowing where to add, but assumed that was too noisy.

If you, the maintainers of this project, like having these and/or other kinds of tests, I’d be happy to help to add more of them and provide usable documentation in other PRs, as desired. Happy also to add a simple github action/badge, as you like.